src/main/java/redis/clients/jedis/BinaryClient.java

```diff
@@ -1244,6 +1256,43 @@ public void moduleUnload(final byte[] name) {
     return args;
   }
 
+  public void aclWhoAmi() { sendCommand(ACL, Keyword.WHOAMI.raw); }
+
+  public void aclGenPass() { sendCommand(ACL, Keyword.GENPASS.raw); }
+
+  public void aclList() { sendCommand(ACL, Keyword.LIST.raw); }
+
+  public void aclUsers() { sendCommand(ACL, Keyword.USERS.raw); }
+
+  public void aclCat() { sendCommand(ACL, Keyword.CAT.raw); }
+
+  public void aclCat(final byte[] category) {
+    sendCommand(ACL, Keyword.CAT.raw, category);
+  }
+
+  public void aclSetUser(final byte[] name) {
+    sendCommand(ACL, Keyword.SETUSER.raw, name);
+  }
+
+  public void aclGetUser(final byte[] name) {
+    sendCommand(ACL, Keyword.GETUSER.raw, name);
+  }
+
+  public void aclSetUser(final byte[] name, byte[][] parameters) {
+    final byte[][] cmdParams = new byte[2 + parameters.length][];
+    int index = 0;
+    cmdParams[index++] = Keyword.SETUSER.raw;
+    cmdParams[index++] = name;
+    for (final byte[] key : parameters) {
+      cmdParams[index++] = key;
+    }
+    sendCommand(ACL, cmdParams);
+  }
+
+  public void aclDelUser(final byte[] name) {
+    sendCommand(ACL, Keyword.DELUSER.raw, name);
+  }
+
```

Comment:
Either move these methods in Client or add respective methods in BinaryJedis.

Reply:
Thanks for your review.
It is not clear to me which class is responsible of what: Client/BinaryCLient vs Jedis/BinaryJedis

how do you choose what goes where?

Reply:
You rightfully placed these methods here. But you did not add respective methods in BinaryJedis. If something is omitted from BinaryJedis, that is also omitted from BinaryClient.

You can check `cluster...` methods for reference.

Reply:
I still do not understand what has to be done, and the role of each the classes. (thanks for your call)

As you have suggested I move the new methods in the interfaces `AdvancedJedisCommands` and `AdvancedBinaryJedisCommands`, but I do now want to do it until I understand "why". ( @gkorland maybe you can help me to understand the flow here)

Reply:
@tgrall I suggested a couple of alternatives including but not limited to:

1. Implement new methods only in Jedis, not in BinaryJedis. This excludes binary support. This way, `BinaryClient` would have *fewest possible new methods*. This follows the path of cluster commands (see `ClusterCommands` interface).

2. Have both binary and non-binary support with two new interfaces e.g. `AccessControlListCommands` and `AccessControlListBinaryCommands`. This follows the path of scripting commands (see `ScriptingCommands`, `BinaryScriptingCommands` interfaces).

3. Have both binary and non-binary support without any new interface just by adding new methods in `AdvancedJedisCommands` and `AdvancedBinaryJedisCommands`. This follows the path of config, slowlog, object, migrate, client, etc commands.This is the one you have mentioned in your last comment.

I prefer option (3). Simply because it supports both binary and non-binary and does not create any new interface. My understanding is that we are perhaps unnecessarily having many interfaces. Just think, what if `JedisCommands` and `BinaryJedisCommands` are split into StringCommands, BinaryStringCommands, ListCommands, BinaryListCommands, HashCommands, BinaryHashCommands, SetCommands, BinarySetCommands, SortedSetCommands, BinarySortedSetCommands, ExpirationCommands, BinaryExpirationCommands, so on and so forth. Feels tiresome? I hope this explains my preference.

Now, you can code the way you want. You are not bound to follow even any of my suggestion. It's just that I am not convinced with your *current* implementation. You are not even bound to convince me either. However, if you don't find any way other than your current one, I would encourage you to choose one from the above three.

Reply:
Thanks @sazzad16, I have implemented the solution (3), not sure how are used the various binary methods. I have added basic tests for them.